modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/item/selector/FDSAdminItemSelector.tsx

```diff
@@ -35,11 +35,31 @@ const FDSAdminItemSelector = ({
 	classNameId,
 	namespace,
 }: {
-	className: String;
-	classNameId: String;
-	namespace: String;
+	className: string;
+	classNameId: string;
+	namespace: string;
 }) => {
-	const [selectedItem, setSelectedItem] = useState<ISelectedItem>();
+	const getSelectedData = () => {
+		const dataset = (window.frameElement as HTMLElement)?.dataset;
+
+		const externalReferenceCode = dataset.selecteditemsercs;
+		const id = dataset.selecteditemsids;
+		const label = dataset.selecteditemslabels;
```

Comment:
The central problem of this task was how to wire selected data to FDS. There are several ways to do it, but I liked the solution from @juanjofgliferay 's previous iteration to use iframe HTML. I only changed to `data-*` attributes, as those better fit the purpose.

Notes:
- The [existing solution](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L541-L622) in `openSelectionModal`, wiring `onOpen`, was always intended to work specifically with search container. It does not fit React. We could force an ugly solution, but there is no need. Iframe attribute is better as we know it'll be set before React resolves.
- Alternative to iframe attributes, that I strongly considered, is adding selection data to URL as parameters. This runs into problems with portlet namespacing, would require more modifications of `layout-content-page-editor`, and could have edge case weird complications. It would also result in more boilerplate, as params need to be handled in backend DisplayContext. It is a viable option is this turn out not acceptable for some reason.